test/integration/managementplane/config_apply_test.go

```diff
@@ -15,143 +15,173 @@ import (
-	})
+	s.Equal(mpi.CommandResponse_COMMAND_STATUS_OK, responses[0].GetCommandResponse().GetStatus())
+	s.Equal("Config apply successful", responses[0].GetCommandResponse().GetMessage())
+	s.Equal(mpi.CommandResponse_COMMAND_STATUS_OK, responses[1].GetCommandResponse().GetStatus())
+	s.Equal("Successfully updated all files", responses[1].GetCommandResponse().GetMessage())
+}
 
-	t.Run("Test 4: File not in allowed directory", func(t *testing.T) {
-		utils.ClearManagementPlaneResponses(t, utils.MockManagementPlaneAPIAddress)
-		utils.PerformInvalidConfigApply(t, nginxInstanceID)
+func (s *ConfigApplyTestSuite) TestConfigApply_Test4_TestInvalidConfig() {
+	err := utils.MockManagementPlaneGrpcContainer.CopyFileToContainer(
+		s.ctx,
+		"../../config/nginx/invalid-nginx.conf",
+		fmt.Sprintf("/mock-management-plane-grpc/config/%s/etc/nginx/nginx.conf", s.nginxInstanceID),
+		0o666,
+	)
+	s.Require().NoError(err)
 
-		responses = utils.ManagementPlaneResponses(t, 1, utils.MockManagementPlaneAPIAddress)
-		t.Logf("Config apply responses: %v", responses)
+	utils.PerformConfigApply(s.T(), s.nginxInstanceID, utils.MockManagementPlaneAPIAddress)
 
-		assert.Equal(t, mpi.CommandResponse_COMMAND_STATUS_FAILURE, responses[0].GetCommandResponse().GetStatus())
-		assert.Equal(t, "Config apply failed", responses[0].GetCommandResponse().GetMessage())
-		assert.Equal(
-			t,
-			"file not in allowed directories /unknown/nginx.conf",
-			responses[0].GetCommandResponse().GetError(),
-		)
-	})
+	responses := utils.ManagementPlaneResponses(s.T(), 2, utils.MockManagementPlaneAPIAddress)
+	s.T().Logf("Config apply responses: %v", responses)
+
+	s.Equal(mpi.CommandResponse_COMMAND_STATUS_ERROR, responses[0].GetCommandResponse().GetStatus())
+	s.Equal("Config apply failed, rolling back config", responses[0].GetCommandResponse().GetMessage())
+	s.Equal(configApplyErrorMessage, responses[0].GetCommandResponse().GetError())
+	s.Equal(mpi.CommandResponse_COMMAND_STATUS_FAILURE, responses[1].GetCommandResponse().GetStatus())
+	s.Equal("Config apply failed, rollback successful", responses[1].GetCommandResponse().GetMessage())
+	s.Equal(configApplyErrorMessage, responses[1].GetCommandResponse().GetError())
+}
+
+func (s *ConfigApplyTestSuite) TestConfigApply_Test5_TestFileNotInAllowedDirectory() {
+	utils.PerformInvalidConfigApply(s.T(), s.nginxInstanceID)
+
+	responses := utils.ManagementPlaneResponses(s.T(), 1, utils.MockManagementPlaneAPIAddress)
+	s.T().Logf("Config apply responses: %v", responses)
+
+	s.Equal(mpi.CommandResponse_COMMAND_STATUS_FAILURE, responses[0].GetCommandResponse().GetStatus())
+	s.Equal("Config apply failed", responses[0].GetCommandResponse().GetMessage())
+	s.Equal(
+		"file not in allowed directories /unknown/nginx.conf",
+		responses[0].GetCommandResponse().GetError(),
+	)
+}
+
+func TestConfigApplyTestSuite(t *testing.T) {
+	suite.Run(t, new(ConfigApplyTestSuite))
```

Comment:
Wondering can the two suites be run from the same function ? 
```suggestion
	suite.Run(t, new(ConfigApplyTestSuite))
	suite.Run(t, new(ConfigApplyChunkingTestSuite))
```